ANMN/burst_averaged_product/burstproduct/burst_average_product.py

```diff
@@ -0,0 +1,401 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+from datetime import datetime
+from file_classifier import MooringFileClassifier
+from generate_netcdf_att import *
+from math import isnan
+from netCDF4 import Dataset, num2date, date2num
+import numpy as np
+import os
+import pandas as pd
+import re
+import shutil
+import tempfile
+import time
+
+"""TODO
+- logging
+"""
+
+
+def get_input_file_rel_path(input_netcdf_file_path):
+    """
+    find the relative path hierarchy of an input FV01 file. The value will be
+    used in a gatt of the burst netcdf file
+    """
+    file_path = MooringFileClassifier.dest_path(input_netcdf_file_path)
+    return file_path
+
+def get_burst_duration(netcdf_file_path):
+    """
+    returns the burst duration as float defined in the NetCDF file if exists
+    """
+    netcdf_file_obj = Dataset(netcdf_file_path, 'r')
+    burst_duration = getattr(netcdf_file_obj, 'instrument_burst_duration', 60.0)
+
+    netcdf_file_obj.close()
+    return burst_duration
+
+def min_bad_qc_flag(varname):
+    """returns the min qc flag value to get rid of per variable"""
+    default_min_qc_flag_value = 3
+    dict_var = {
+        'CHLF':     4,
+        'CHLR':     4,
+        'CHLU':     4,
+        'CPHL':     4,
+        'DOX1_2':   3,
+        'DOX2':     3,
+        'FLU2':     4,
+        'FLNTU':    4,
+        'PRES_REL': 4,
+        'PSAL':     4,
+        'TEMP':     4,
+        'TURB':     4,
+    }
+
+    if varname in dict_var.keys():
+        return dict_var[varname]
+    else:
+        return default_min_qc_flag_value
+
+def create_burst_average_var(netcdf_file_path):
+    """
+    create burst data from all vars available in netcdf
+    """
+    time_values              = get_time_val(netcdf_file_path)
+    varname_to_burst_average = list_var_not_to_average(netcdf_file_path)
+    burst_duration           = get_burst_duration(netcdf_file_path)
+    burst_vars               = {}
+
+    for varname in varname_to_burst_average:
+        [var_values, var_qc_flag_exclusion] = get_var_val_var_qc_exclusion(netcdf_file_path, varname, min_bad_qc_flag(varname))
```

Comment:
You don't need `[]` to unpack a tuple, i.e. `var_values, var_qc_flag_exclusion = ...` does exactly the same thing.

Also, it would be nice to wrap such long lines to something more legible, e.g.
```python
        var_values, var_qc_flag_exclusion = get_var_val_var_qc_exclusion(
            netcdf_file_path, varname, min_bad_qc_flag(varname)
        )
```
(See [style guide](https://www.python.org/dev/peps/pep-0008/#id17))